src/main.ts

```diff
@@ -1,55 +1,24 @@
-import * as core from "@actions/core";
-import * as github from "@actions/github";
-import { Octokit } from "@octokit/rest";
-import { Value } from "@sinclair/typebox/value";
-import { envSchema, pluginSettingsSchema, PluginInputs, pluginSettingsValidator } from "./types";
+import { createActionsPlugin } from "@ubiquity-os/plugin-sdk";
+import { createAdapters } from "./adapters";
+import { SupportedEvents } from "./types/context";
+import { Env, envSchema } from "./types/env";
+import { PluginSettings, pluginSettingsSchema } from "./types/plugin-input";
+import { Command } from "./types/command";
 import { plugin } from "./plugin";
-
-/**
- * How a GitHub action executes the plugin.
- */
-export async function run() {
-  const payload = github.context.payload.inputs;
-
-  const env = Value.Decode(envSchema, payload.env);
-  const settings = Value.Decode(pluginSettingsSchema, Value.Default(pluginSettingsSchema, JSON.parse(payload.settings)));
-
-  if (!pluginSettingsValidator.test(settings)) {
-    throw new Error("Invalid settings provided");
+import { LogLevel } from "@ubiquity-os/ubiquity-os-logger";
+
+createActionsPlugin<PluginSettings, Env, Command, SupportedEvents>(
```

Comment:
```suggestion
createActionsPlugin<PluginSettings, Env, null, SupportedEvents>(
```
you can put `null` if there's no command